File: src/main/java/com/kelsos/mbrc/commands/InitiateConnectionCommand.java
package com.kelsos.mbrc.commands;

import com.google.inject.Inject;
import com.kelsos.mbrc.enums.SocketAction;
import com.kelsos.mbrc.interfaces.ICommand;
import com.kelsos.mbrc.interfaces.IEvent;
import com.kelsos.mbrc.services.SocketService;

public class InitiateConnectionCommand implements ICommand
{
	@Inject
	SocketService socketService;

	@Override
	public void execute(IEvent e)
	{
		socketService.SocketManager(SocketAction.START);
	}
}

File: app/src/main/java/com/kelsos/mbrc/commands/InitiateConnectionCommand.kt
package com.kelsos.mbrc.commands

import com.kelsos.mbrc.annotations.SocketAction.START
import com.kelsos.mbrc.networking.client.SocketService
import com.kelsos.mbrc.networking.protocol.ProtocolAction
import com.kelsos.mbrc.networking.protocol.ProtocolMessage

class InitiateConnectionCommand(
  private val socketService: SocketService,
) : ProtocolAction {
  override fun execute(message: ProtocolMessage) {
    socketService.socketManager(START)
  }
}
